vue2-vuetify/src/complex/components/AdditionalProperties.vue

```diff
@@ -0,0 +1,463 @@
+        /></v-col>
+        <v-col v-if="control.enabled" class="shrink">
+          <v-tooltip bottom>
+            <template v-slot:activator="{ on: onTooltip }">
+              <v-btn
+                v-on="onTooltip"
+                fab
+                text
+                elevation="0"
+                small
+                :aria-label="deleteLabel"
+                :disabled="
+                  !control.enabled ||
+                  (appliedOptions.restrict &&
+                    control.schema.minProperties !== undefined &&
+                    control.data &&
+                    Object.keys(control.data).length <=
+                      control.schema.minProperties)
+                "
+                @click="removeProperty(element.propertyName)"
+              >
+                <v-icon class="notranslate">mdi-delete</v-icon>
+              </v-btn>
+            </template>
+            {{ deleteLabel }}
+          </v-tooltip></v-col
+        >
+      </v-row>
+    </v-container>
+  </v-card>
+</template>
+
+<script lang="ts">
+import {
+  composePaths,
+  createControlElement,
+  createDefaultValue,
+  encode,
+  Generate,
+  getI18nKey,
+  GroupLayout,
+  JsonSchema,
+  JsonSchema7,
+  UISchemaElement,
+  validate,
+} from '@jsonforms/core';
+import {
+  DispatchRenderer,
+  useJsonFormsControlWithDetail,
+} from '@jsonforms/vue2';
+import Ajv, { ValidateFunction } from 'ajv';
+import get from 'lodash/get';
+import isPlainObject from 'lodash/isPlainObject';
+import startCase from 'lodash/startCase';
+import { defineComponent, PropType, Ref, ref } from 'vue';
+import {
+  VBtn,
+  VCard,
+  VCardTitle,
+  VCol,
+  VContainer,
+  VHover,
+  VIcon,
+  VRow,
+  VSpacer,
+  VTextField,
+  VToolbar,
+  VToolbarTitle,
+  VTooltip,
+} from 'vuetify/lib';
+import { DisabledIconFocus } from '../../controls/directives';
+import { useStyles } from '../../styles';
+import { useAjv, useControlAppliedOptions, useTranslator } from '../../util';
+
+type Input = ReturnType<typeof useJsonFormsControlWithDetail>;
+interface AdditionalPropertyType {
+  propertyName: string;
+  path: string;
+  schema: JsonSchema | undefined;
+  uischema: UISchemaElement | undefined;
+}
+
+const reuseAjvForSchema = (ajv: Ajv, schema: JsonSchema): Ajv => {
+  if (
+    Object.prototype.hasOwnProperty.call(schema, 'id') ||
+    Object.prototype.hasOwnProperty.call(schema, '$id')
+  ) {
+    ajv.removeSchema(schema);
+  }
+  return ajv;
+};
+
+export default defineComponent({
+  name: 'additional-properties',
+  components: {
+    DispatchRenderer,
+    VCard,
+    VTooltip,
+    VToolbar,
+    VIcon,
+    VBtn,
+    VCardTitle,
+    VSpacer,
+    VToolbarTitle,
+    VTextField,
+    VContainer,
+    VRow,
+    VCol,
+    VHover,
+  },
+  directives: {
+    DisabledIconFocus,
+  },
+  props: {
+    input: {
+      type: Object as PropType<Input>,
+      required: true,
+    },
+  },
+  setup(props) {
+    const control = props.input.control as any as Ref<
```

Comment:
Why do you need to cast this here? Couldn't you instead adapt the type of the prop instead?

Reply:
@kchobantonov Please have a look again at this comment. I think it was hidden earlier.

Reply:
because the control is a reference and as such need to be accessed with .value - the IDE just doesn't know that the actual value that is going to be passed is not of type T but Ref<T> and if I do not do that type cast then the rest of the code the IDE will not give me an error if I access the properties directly instead via the .value property as it should when I'm in the setup function - the Vue does not automatically do the property access for ref objects in the setup like that is happening in the methods, computed values or the template.